src/Unity.Mathematics/Geometry/MinMaxAABB.cs

```diff
@@ -1,49 +1,332 @@
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator AABB(MinMaxAABB aabb)
         {
-            return new AABB { Center = (aabb.Min + aabb.Max) * 0.5F, Extents = (aabb.Max - aabb.Min) * 0.5F};
+            // AABB Extents are actually half extents.  The full extents in AABB is called Size.
+            return new AABB { Center = aabb.Center, Extents = aabb.HalfExtents };
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool Equals(MinMaxAABB other)
         {
             return Min.Equals(other.Min) && Max.Equals(other.Max);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public override string ToString()
+        {
+            return string.Format("MinMaxAABB({0}, {1})", Min, Max);
+        }
+    }
+
+    internal static partial class Math
+    {
+        /// <summary>
+        /// Transforms the AABB with the given transform.
+        /// </summary>
+        /// <remarks>
+        /// The resulting AABB encapsulates the transformed AABB which may not be axis aligned after the transformation.
+        /// </remarks>
+        /// <param name="transform">Transform to apply to AABB.</param>
+        /// <param name="aabb">AABB to be transformed.</param>
+        /// <returns>Transformed AABB.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static MinMaxAABB Transform(RigidTransform transform, MinMaxAABB aabb)
+        {
+            float3 halfExtentsInA = aabb.HalfExtents;
+
+            // Rotate each axis individually and find their new positions in the rotated space.
+            float3 x = math.rotate(transform.rot, new float3(halfExtentsInA.x, 0, 0));
+            float3 y = math.rotate(transform.rot, new float3(0, halfExtentsInA.y, 0));
+            float3 z = math.rotate(transform.rot, new float3(0, 0, halfExtentsInA.z));
+
+            // Find the new max corner by summing the rotated axes.  Absolute value of each axis
+            // since we are trying to find the max corner.
+            float3 halfExtentsInB = math.abs(x) + math.abs(y) + math.abs(z);
+            float3 centerInB = math.transform(transform, aabb.Center);
+
+            return new MinMaxAABB(centerInB - halfExtentsInB, centerInB + halfExtentsInB);
+        }
+
+        /// <summary>
+        /// Transforms the AABB with the given transform.
+        /// </summary>
+        /// <remarks>
+        /// The resulting AABB encapsulates the transformed AABB which may not be axis aligned after the transformation.
+        /// </remarks>
+        /// <param name="transform">Transform to apply to AABB.</param>
+        /// <param name="aabb">AABB to be transformed.</param>
+        /// <returns>Transformed AABB.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static MinMaxAABB Transform(float4x4 transform, MinMaxAABB aabb)
```

Comment:
Not sure about how you are addressing this in general, but given the performance cost on Mono - would it make sense to pass `transform` by reference?

Reply:
It would make sense to do so.  I haven't been doing this in Mathematics yet because burst actually breaks on Mathematics functions when using pass by ref, but that should only apply to intrinsic functions in Mathematics.  Ref in this context _should_ be safe, so I'll try it out and if it looks OK, I'll make the change.

Reply:
Sigh.  It works locally for me but on CI, using `in` parameters fails, so I think I'm going to back away from using `in` for this PR and address it in a subsequent change (see DOTS-1600).